src/util/rewriter.js

```diff
@@ -0,0 +1,146 @@
+const assert = require('assert');
+const objectScan = require('object-scan');
+const objectFields = require('object-fields');
+const sortFn = require('../util/sort-fn');
+const { pluginTypes } = require('./plugin');
+
+const defineProperty = (target, k, v) => Object.defineProperty(target, k, { value: v, writable: false });
+const SORT_VALUE = Symbol('sortValue');
+const setSortValue = (input, value) => defineProperty(input, SORT_VALUE, value);
+const getSortValue = input => input[SORT_VALUE];
+
+const compileTargetToCallback = (type, plugins) => {
+  assert(plugins.every(p => p.type === type));
+
+  const targetToPlugins = plugins
+    .reduce((prev, plugin) => {
+      const target = plugin.target.endsWith('.') ? plugin.target.slice(0, -1) : plugin.target;
+      const key = type === 'INJECT' ? target.split('.').slice(0, -1).join('.') : target;
+      if (prev[key] === undefined) {
+        // eslint-disable-next-line no-param-reassign
+        prev[key] = [];
```

Comment:
Is it just for readability that it's not
```suggestion
        Object.assign(prev, { [key]: [] });
```
?

Reply:
fixed

Reply:
changed it